File: problems/problems_111/solution.go
package problem111

import (
	. "leetCode/golang/models"
	"math"
	"strings"
)

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func minDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	if root.Left == nil && root.Right == nil {
		return 1
	}
	ans := math.MaxInt32
	if root.Left != nil {
		ans = min(ans, minDepth(root.Left)+1)
	}
	if root.Right != nil {
		ans = min(ans, minDepth(root.Right)+1)
	}
	return ans
}

func Solve(input string) any {
	values := strings.Split(input, "\n")
	var root *TreeNode

	root = ArrayToTree(values[0])

	return minDepth(root)
}

File: problems/problems_111/solution.ts
import {JSONArrayToTreeNode,TreeNode} from "../../typescript/models/treenode";

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function minDepth(root: TreeNode | null): number {
	if (root == null) {
		return 0;
	}
	if (root.left == null && root.right == null) {
		return 1;
	}
	let ans: number = 0x3f3f3f;
	if (root.left != null) {
		ans = Math.min(ans, minDepth(root.left) + 1);
	}
	if (root.right != null) {
		ans = Math.min(ans, minDepth(root.right) + 1);
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const splits: string[] = inputJsonElement.split("\n");
	const root: TreeNode | null = JSONArrayToTreeNode(JSON.parse(splits[0]));
	return minDepth(root);
}
